packages/opencensus-resource-util/src/resource-utils.ts

```diff
@@ -0,0 +1,204 @@
+  } catch {
+    return false;
+  }
+}
+
+/** Returns Resource for GCP GCE instance. */
+export async function getComputerEngineResource(): Promise<Resource> {
+  if (Object.keys(gceResourceLabels).length === 0) {
+    const [projectId, instanceId, zoneId] =
+        await Promise.all([getProjectId(), getInstanceId(), getZone()]);
+    gceResourceLabels[constants.GCP_ACCOUNT_ID_KEY] = projectId;
+    gceResourceLabels[constants.GCP_INSTANCE_ID_KEY] = instanceId;
+    gceResourceLabels[constants.GCP_ZONE_KEY] = zoneId;
+  }
+  return {type: constants.GCP_GCE_INSTANCE_TYPE, labels: gceResourceLabels};
+}
+
+/** Returns Resource for GCP GKE container. */
+export async function getKubernetesEngineResource(): Promise<Resource> {
+  if (Object.keys(gkeResourceLabels).length === 0) {
+    const [projectId, zoneId, clusterName, hostname] = await Promise.all(
+        [getProjectId(), getZone(), getClusterName(), getHostname()]);
+    gkeResourceLabels[constants.GCP_ACCOUNT_ID_KEY] = projectId;
+    gkeResourceLabels[constants.GCP_ZONE_KEY] = zoneId;
+    gkeResourceLabels[constants.K8S_CLUSTER_NAME_KEY] = clusterName;
+    gkeResourceLabels[constants.K8S_NAMESPACE_NAME_KEY] =
+        process.env.NAMESPACE || '';
+    gkeResourceLabels[constants.K8S_POD_NAME_KEY] = hostname;
+    gkeResourceLabels[constants.K8S_CONTAINER_NAME_KEY] =
+        process.env.CONTAINER_NAME || '';
+  }
+  return {type: constants.K8S_CONTAINER_TYPE, labels: gkeResourceLabels};
+}
+
+/** Returns Resource for AWS EC2 instance. */
+export async function getAwsEC2Resource(): Promise<Resource> {
+  return {type: constants.AWS_EC2_INSTANCE_TYPE, labels: awsResourceLabels};
+}
+
+/**
+ * Establishes an HTTP connection to AWS instance identity document url.
+ * If the application is running on an EC2 instance, we should be able
+ * to get back a valid JSON document. Parses that document and stores
+ * the identity properties in a local map.
+ */
+async function awsMetadataAccessor<T>(): Promise<T> {
+  return new Promise((resolve, reject) => {
+    const req = http.get(AWS_INSTANCE_IDENTITY_DOCUMENT_URI, (res) => {
+      const {statusCode} = res;
+      res.setEncoding('utf8');
+      let rawData = '';
+      res.on('data', (chunk) => rawData += chunk);
+      res.on('end', () => {
+        if (statusCode && statusCode >= 200 && statusCode < 300) {
+          try {
+            resolve(JSON.parse(rawData));
+          } catch (e) {
+            res.resume();  // consume response data to free up memory
+            reject(e);
+          }
+        } else {
+          res.resume();  // consume response data to free up memory
+          reject(new Error('Failed to load page, status code: ' + statusCode));
+        }
+      });
+    });
+    req.on('error', (err) => reject(err));
+  });
+}
+
+/** Gets project id from GCP project metadata. */
+async function getProjectId() {
+  try {
+    return await gcpMetadata.project('project-id');
+  } catch {
+    /* ignore */
+  }
+  return '';
```

Comment:
What would you think about moving this `return ''` up into the `catch` block? Then you could potentially remove the `/* ignore */` comment since the error handling just causes an empty string to return.

Reply:
Good call. I was thinking about it, but missed somehow. done.